sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -181,6 +181,27 @@ case class DataSourceResolution(
       val aliased = delete.tableAlias.map(SubqueryAlias(_, relation)).getOrElse(relation)
       DeleteFromTable(aliased, delete.condition)
 
+    case ShowTablesStatement(None, pattern) =>
+      defaultCatalog match {
+        case Some(_) =>
+          throw new AnalysisException(
+            "The current namespace is not available in v2 catalog yet")
```

Comment:
Can we wait a little bit for https://github.com/apache/spark/pull/25368 ? With it, we can access the current namespace from `CatalogManager` in this rule.

Reply:
it has been merged now.

Reply:
Thanks @cloud-fan. Can you check https://github.com/apache/spark/pull/25368/files#r315980292? I am not sure if the `CatalogManager._currentNamespace`'s behavior is correct.

Reply:
Now this is fixed.